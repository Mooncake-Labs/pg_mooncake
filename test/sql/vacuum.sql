DROP EXTENSION IF EXISTS pg_mooncake CASCADE;
CREATE EXTENSION pg_mooncake;

CREATE TABLE t (a int, b int) USING columnstore;
INSERT INTO t VALUES (1, 1);
INSERT INTO t VALUES (2, 2);
INSERT INTO t VALUES (3, 3);
INSERT INTO t VALUES (4, 4);
INSERT INTO t VALUES (5, 5);

SELECT COUNT(*) FROM mooncake.data_files;

VACUUM t;

SELECT COUNT(*) FROM mooncake.data_files;

DROP TABLE t;
