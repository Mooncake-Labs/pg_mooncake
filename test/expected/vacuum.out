DROP EXTENSION IF EXISTS pg_mooncake CASCADE;
CREATE EXTENSION pg_mooncake;
CREATE TABLE t (a int, b int) USING columnstore;
INSERT INTO t VALUES (1, 1);
INSERT INTO t VALUES (2, 2);
INSERT INTO t VALUES (3, 3);
INSERT INTO t VALUES (4, 4);
INSERT INTO t VALUES (5, 5);
SELECT COUNT(*) FROM mooncake.data_files;
 count 
-------
     5
(1 row)

VACUUM t;
SELECT COUNT(*) FROM mooncake.data_files;
 count 
-------
     1
(1 row)

SELECT * FROM t ORDER BY a;
 a | b 
---+---
 1 | 1
 2 | 2
 3 | 3
 4 | 4
 5 | 5
(5 rows)

DROP TABLE t;
